api/routes/user.py

```diff
@@ -18,3 +21,59 @@ def get_current_user_profile(
 
     log(log.INFO, f"User {current_user.username} requested his profile")
     return current_user
+
+
+@user_router.patch("/me/language", status_code=status.HTTP_200_OK, response_model=s.User)
+def update_user_language(
+    language_update: s.LanguageUpdate,
+    current_user: m.User = Depends(get_current_user),
+    db: Session = Depends(get_db),
+):
+    """Updates the current user's language"""
+    if current_user.language != language_update.language:
+        current_user.language = language_update.language
+
+    db.add(current_user)
+    db.commit()
+    db.refresh(current_user)
+
+    log(log.INFO, f"User {current_user.username} updated his language to {current_user.language}")
+    return current_user
+
+
+@user_router.patch("/me/username", status_code=status.HTTP_200_OK, response_model=s.User)
+def update_user_username(
+    username_update: s.UsernameUpdate,
+    current_user: m.User = Depends(get_current_user),
+    db: Session = Depends(get_db),
+):
+    """Updates the current user's username"""
+    if current_user.username != username_update.username:
+        current_user.username = username_update.username
+
+    db.add(current_user)
+    db.commit()
+    db.refresh(current_user)
+
+    log(log.INFO, f"User {current_user.username} updated his username to {current_user.username}")
+    return current_user
+
+
+@user_router.delete("/me", status_code=status.HTTP_204_NO_CONTENT)
+def delete_user(
+    current_user: m.User = Depends(get_current_user),
+    db: Session = Depends(get_db),
+):
+    """Deletes the current user"""
+
+    if current_user.is_deleted:

```

Comment:
I think this logic should be part of `get_current_user` function. If user is deleted, he should not be able to perform any actions, which require authentication and/or authorization

Reply:
Also, please check user authentication function to make sure that deleted users cannot login